main/src/ca/uwaterloo/flix/api/lsp/provider/InlayHintProvider.scala

```diff
@@ -15,13 +15,33 @@
  */
 package ca.uwaterloo.flix.api.lsp.provider
 
-import ca.uwaterloo.flix.api.lsp.{InlayHint, Range}
+import ca.uwaterloo.flix.api.lsp.acceptors.{FileAcceptor}
+import ca.uwaterloo.flix.language.ast.shared.SymUse
+import ca.uwaterloo.flix.language.ast.TypedAst.{Expr, Root}
+import ca.uwaterloo.flix.api.lsp.{Consumer, InlayHint, InlayHintKind, Position, Range}
+import ca.uwaterloo.flix.api.lsp.Visitor
 
 object InlayHintProvider {
+  def getInlayHints(uri: String, range: Range)(implicit root: Root): List[InlayHint] = {
+    var inlayHints: List[InlayHint] = Nil
+    object opSymUseConsumer extends Consumer {
```

Comment:
I would separate the two concerns: finding where to create an inline hint and creating the InlayHint. 

Here you have them in the same code. But they are really different things. 

Here is an example: https://github.com/flix/flix/blob/master/main/test/ca/uwaterloo/flix/api/lsp/TestCompletionProvider.scala#L589

Reply:
Sure close, but opSymUses.foreach can be in its own function too right?